### PR TITLE
Remove unnecessay call to findNewReplicaSet

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR removes the unneeded call to findNewReplicaSet for findOldReplicaSets.

```release-note
NONE
```

### CODE REVIEW

**Comment by apelisse** [comment]
Why is this needed, how did you find out?

**Comment by tedyu** [comment]
@apelisse
I was reading this part of code and found the redundant code.
The extra call can be removed as shown.

What do you think ?

**Comment by apelisse** [comment]
I don't know this part of the code and what the implications are. 

/assign @janetkuo

**Comment by tedyu** [comment]
@mengqiy
Can you take a look as well ?

File: staging/src/k8s.io/kubectl/pkg/util/deployment/deployment.go [hunk 1]
```diff
@@ -181,10 +181,9 @@ func (o replicaSetsByCreationTimestamp) Less(i, j int) bool {
 
 // // FindOldReplicaSets returns the old replica sets targeted by the given Deployment, with the given slice of RSes.
 // // Note that the first set of old replica sets doesn't include the ones with no pods, and the second set of old replica sets include all old replica sets.
-func findOldReplicaSets(deployment *appsv1.Deployment, rsList []*appsv1.ReplicaSet) ([]*appsv1.ReplicaSet, []*appsv1.ReplicaSet) {
+func findOldReplicaSets(deployment *appsv1.Deployment, rsList []*appsv1.ReplicaSet, newRS *appsv1.ReplicaSet) ([]*appsv1.ReplicaSet, []*appsv1.ReplicaSet) {
```
Comment (dixudx) [inline staging/src/k8s.io/kubectl/pkg/util/deployment/deployment.go]: I don't think we need to update this interface. It is unnecessary.

Comment (tedyu) [reply staging/src/k8s.io/kubectl/pkg/util/deployment/deployment.go]: findOldReplicaSets is an un-exported func.
newRS is needed by this func.

Can you clarify how newRS is passed to this func without changing the func ?

Comment (dixudx) [reply staging/src/k8s.io/kubectl/pkg/util/deployment/deployment.go]: I mean just keeping current two methods as they are. They are both local methods. I think it's okay to keep them unchanged.

I am open for the changes if you prefer the newer interface.

Comment (tedyu) [reply staging/src/k8s.io/kubectl/pkg/util/deployment/deployment.go]: The newer interface calculates newRS once.
I prefer the newer interface.

**Comment by tedyu** [comment]
@janetkuo 
Please review

**Comment by tedyu** [comment]
ping @janetkuo for review

**Review by janetkuo — APPROVED** [review]

**Comment by tedyu** [comment]
@janetkuo 
```
Needs approved label
```
Can you say /approve ?

Thanks

**Comment by janetkuo** [comment]
/lgtm 

This doesn't change any logic, and simply reduces a function call to get new RS. 

@seans3 This is copied from deployment controller. What's the plan going forward for code changes in these two places?

**Comment by tedyu** [comment]
@seans3
Mind taking a look at the above question ?

**Comment by tedyu** [comment]
@janetkuo 
Can you approve this ?

**Comment by tedyu** [comment]
ping @janetkuo for approval

**Comment by tedyu** [comment]
@janetkuo 
Can you approve ?

**Comment by janetkuo** [comment]
The best practice is to have a separate person to approve to have a second eye.

**Pull request merged**